drivers/wifi/esp8266/Kconfig.esp8266

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2019 Tobias Svehagen
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig WIFI_ESP8266
+	bool "Espressif esp8266 support"
+	select MODEM
+	select MODEM_CONTEXT
+	select MODEM_CMD_HANDLER
+	select MODEM_IFACE_UART
```

Comment:
This really looks like we should separate the AT command handling from modem driver into a library, it is just weird to have modem stuff in wifi driver.

Reply:
@jukkar That could be a good idea as we will have more radios using AT commands via UART, another example would be LoRA/SigFox.

On the other hand one could argue that as they are using AT commands they are in fact emulating modems.

Reply:
I think that could be a good idea. I'd like to hear @mike-scott's opinion on that.

Reply:
@jukkar Is this a problem for getting this merged?

Reply:
> Is this a problem for getting this merged?

No, just a note for future development.